### PR TITLE
Fix line comment after `as`/`satisfies` moving past the type and semicolon

### PR DESCRIPTION
**Pull request by koreahghg**
## Description

A single-line comment placed right after `as`/`satisfies`, with the type on the next line, gets relocated all the way past the type and the statement's semicolon on the very first format:

```ts
const x = value satisfies // comment
Foo;
```

formats to:

```ts
const x = value satisfies Foo; // comment
```

The equivalent block comment (`satisfies /* comment */\nFoo;`) is handled correctly and stays anchored to the type, so this is inconsistent as well as surprising.

**Root cause:** `handleBinaryCastExpressionComment` in `src/language-js/comments/handle-comments.js` reattaches a comment between the cast expression's value and its type as a leading comment on the type — but it explicitly excludes single-line comments (`!isSingleLineComment(...)`). For `endOfLine` placement (comment trails code on the same line with nothing after it), excluding it means the comment falls through to the default attachment: a trailing comment on the preceding expression, deferred via line suffix to the next hardline. Since there's no hardline until after the whole statement, the comment surfaces past the type and semicolon.

This is a narrower, previously-unfixed variant of the same area #19939 touched: that PR fixed the case where the comment is already on its own separate line before the type (`ownLine` placement); this fixes the case where the comment is glued to `as`/`satisfies` on the same line (`endOfLine` placement).

**Fix:** only reattach single-line comments for `endOfLine` placement, so they land as a leading comment on the type (matching how an own-line comment in the same position is already handled):

```ts
const x = value satisfies
  // comment
  Foo;
```

`remaining`/`ownLine` placements (e.g. `{} /* comment */ satisfies {}`, block comments already on their own line) are untouched — verified against the full `tests/format` suite (34,537 tests) plus `FULL_TEST=1` (AST-equivalence, idempotency) on the affected directories, with only the one pre-existing snapshot that exercised this exact scenario changing to the fixed output.

> Note: this fix was investigated and implemented with Claude Code's assistance (root-cause tracing through the comment-attachment pipeline, edge-case testing, and regression verification), then reviewed by me before submitting.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the contributing guidelines.
- [ ] I did _not_ use AI to generate this PR.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### CODE REVIEW

File: src/language-js/comments/handle-comments.js [hunk 1]
```diff
@@ -1125,12 +1125,14 @@ function handleBinaryCastExpressionComment({
...
   text,
+  placement,
 }) {
   // Avoid break before `as` and `satisfies`
   if (
     isBinaryCastExpression(enclosingNode) &&
     precedingNode === enclosingNode.expression &&
-    !isSingleLineComment(comment, text)
+    // `endOfLine` single-line comments still need reattaching, otherwise they get deferred past the type and semicolon
+    (!isSingleLineComment(comment, text) || placement === "endOfLine")
```
Comment (fisker) [inline src/language-js/comments/handle-comments.js:1135]: Which case will be broken if we remove this line?

Comment (koreahghg) [reply src/language-js/comments/handle-comments.js:1135]: SPAM, can't read. [Edit by @fisker]

Comment (fisker) [reply src/language-js/comments/handle-comments.js:1135]: Please answer this question and determine if it should be removed yourself, not forward it to your LLM.

Comment (koreahghg) [reply src/language-js/comments/handle-comments.js:1135]: I simplified it:

```js
!(isSingleLineComment(comment, text) && placement === "remaining")
```

This is equivalent (`ownLine` placement behaves the same either way, since the default attachment already does the same thing this handler would), but reads closer to the actual intent: only exclude a single-line comment when it's in `remaining` placement (e.g. `{} /* c */ satisfies {}`, `1 as /*\nc\n*/const;`), where the default tie-break/fallback already does the right thing. `endOfLine` single-line comments (the bug this PR fixes) and `ownLine` ones both go through this handler now.

Ran the full `tests/format` suite (34,537 tests) plus `FULL_TEST=1` on the affected directories  all pass with no snapshot changes.

Comment (fisker) [reply src/language-js/comments/handle-comments.js:1135]: Sorry, I don't want to talk to a machine
